### PR TITLE
89920 Add upgrade CLI for diff providers

### PR DESCRIPTION
**Pull request by mmcelroy91**
## Jira Ticket

https://jira.d2iq.com/browse/D2IQ-89920?filter=-1
https://jira.d2iq.com/browse/D2IQ-88470?filter=-1

## Description of changes being made
Additional instructions for upgrade DKP level documentation both Essential and Enterprise depending on Infrastructure Provider.  i.e. GCP, AWS, vSphere, Azure, AKS, EKS, preprovisioned by listing them and explaining to change in each CLI for Upgrade the control plane using DKP CLI - dkp update controlplane eks
Upgrade the worker node pools using DKP CLI - dkp update nodepool eks
Upgrade the addons - dkp update addons eks

and a couple of minor wording changes hitch-hiked on this one.
### Preview

You can preview the docs build using the following URL, adding the PR # where specified:
http://docs-d2iq-com-pr-4508.s3-website-us-west-2.amazonaws.com/

## Checklist

- [ ] Test all commands and procedures, if applicable.
- [ ] Update all links if you are moving a page.
- [ ] Add release date to Release Notes page in the following format: <Package> was released on <Day>, <Month> <Year> Example: `Mesosphere® DC/OS™ 2.1.0 was released on 9, June 2020`

See the [contribution guidelines](https://github.com/mesosphere/dcos-docs-site/blob/main/CONTRIBUTING.md) for more information.

### CODE REVIEW

File: pages/dkp/konvoy/2.3/choose-infrastructure/eks/advanced/new/index.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ enterprise: false
 
 Before you start, make sure you have completed the steps in [Bootstrap][bootstrap].
 
-## Create a new AWS Kubernetes cluster
+## Create a new EKS Kubernetes cluster
```
Comment (msdolbey) [inline pages/dkp/konvoy/2.3/choose-infrastructure/eks/advanced/new/index.md:12]: oh my

File: pages/dkp/konvoy/2.3/choose-infrastructure/gcp/quick-start-gcp/index.md [hunk 2]
```diff
@@ -20,10 +20,11 @@ Before starting the Konvoy installation, verify that you have:
 - [kubectl][install_kubectl] for interacting with the running cluster.
 - A valid GCP account with [credentials configured](https://github.com/kubernetes-sigs/cluster-api-provider-gcp).
 
-# Create a Cluster in Google Cloud Platform using clusterctl
+# Create a Cluster in Google Cloud Platform 
 <!NEED CONFIRMATION OF STEPS PRE-RELEASE>
 ## Setup your gcloud CLI
 
+The following steps will be performed using clusterctl.
```
Comment (msdolbey) [inline pages/dkp/konvoy/2.3/choose-infrastructure/gcp/quick-start-gcp/index.md:27]: wild that this isn't in the accept.txt manifest yet, I can add it super quick

File: pages/dkp/konvoy/2.3/dkp-upgrade/upgrade-konvoy/enterprise/index.md [hunk 3]
```diff
@@ -101,7 +105,7 @@ Your cluster comes preconfigured with a few different core addons that provide f
 
 <p class="message--warning"><strong>IMPORTANT:</strong>Ensure your <code>dkp</code> configuration references the management cluster where you want to run the upgrade by setting the <code>KUBECONFIG</code> environment variable, or using the <code>--kubeconfig</code> flag, <a href="https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/">in accordance with Kubernetes conventions</a>.
 
-Upgrade the core addons in a cluster using the 'dkp upgrade addons' command specifying the cluster infrastructure (choose [aws, azure, vsphere, gcp, preprovisioned]) and the name of the cluster.
+Upgrade the core addons in a cluster using the 'dkp upgrade addons' command specifying the cluster infrastructure (choose [aws, azure, vsphere, gcp, eks, preprovisioned]) and the name of the cluster.
```
Comment (msdolbey) [inline pages/dkp/konvoy/2.3/dkp-upgrade/upgrade-konvoy/enterprise/index.md:108]:
```suggestion
Upgrade the core addons in a cluster using the `dkp upgrade addons` command specifying the cluster infrastructure (choose `[aws, azure, vsphere, gcp, eks, preprovisioned]`) and the name of the cluster.
```

File: pages/dkp/konvoy/2.3/dkp-upgrade/upgrade-konvoy/enterprise/index.md [hunk 4]
```diff
@@ -160,27 +164,27 @@ When upgrading the Kubernetes version of a cluster, first upgrade the control pl
     - If an AMI was specified when initially creating a cluster for AWS, you must build a new one with <a href="/dkp/konvoy/2.3/image-builder/create-ami/">Konvoy Image Builder</a>.
     - If an Azure Machine Image was specified for Azure, you must build a new one with <a href="/dkp/konvoy/2.3/image-builder/create-azure-image/">Konvoy Image Builder</a>.
 
-2. Upgrade the Kubernetes version of the control plane.
+2. Upgrade the Kubernetes version of the control plane. As shown in update addons section above, replace infrastructure provider name in the command [aws, azure, vsphere, gcp, eks, preprovisioned].
```
Comment (msdolbey) [inline pages/dkp/konvoy/2.3/dkp-upgrade/upgrade-konvoy/enterprise/index.md:167]:
```suggestion
2. Upgrade the Kubernetes version of the control plane. As shown in update addons section above, replace infrastructure provider name in the command [`aws`, `azure`, `vsphere`, `gcp`, `eks`, `preprovisioned`].
```

File: pages/dkp/konvoy/2.3/dkp-upgrade/upgrade-konvoy/enterprise/index.md [hunk 5]
```diff
@@ -160,27 +164,27 @@ When upgrading the Kubernetes version of a cluster, first upgrade the control pl
...
+The output should be similar to the below example with the provider name corresponding to the one you entered in the command line:
 
 ```text
 Updating control plane resource controlplane.cluster.x-k8s.io/v1beta1, Kind=KubeadmControlPlane default/my-aws-cluster-control-plane
 Waiting for control plane update to finish.
  ✓ Updating the control plane
 ```
     
-3. Upgrade the Kubernetes version of each of your node pools. Replace `my-nodepool` with the name of the node pool.
+3. Upgrade the Kubernetes version of each of your node pools. Replace `my-nodepool` with the name of the node pool. Also, replace the name of the infrastructure provider in the command [aws, azure, vsphere, gcp, eks, preprovisioned].
```
Comment (msdolbey) [inline pages/dkp/konvoy/2.3/dkp-upgrade/upgrade-konvoy/enterprise/index.md:181]:
```suggestion
3. Upgrade the Kubernetes version of each of your node pools. Replace `my-nodepool` with the name of the node pool. Also, replace the name of the infrastructure provider in the command [`aws`, `azure`, `vsphere`, `gcp`, `eks`, `preprovisioned`].
```

File: pages/dkp/konvoy/2.3/dkp-upgrade/upgrade-konvoy/essential/index.md [hunk 6]
```diff
@@ -96,7 +100,7 @@ Your cluster comes preconfigured with a few different core addons that provide f
 
 <p class="message--warning"><strong>IMPORTANT:</strong>If you have more than one essential cluster, ensure your <code>dkp</code> configuration references the management cluster where you want to run the upgrade by setting the <code>KUBECONFIG</code> environment variable, or using the <code>--kubeconfig</code> flag, <a href="https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/">in accordance with Kubernetes conventions</a>.
 
-Upgrade the core addons in a cluster using the 'dkp upgrade addons' command specifying the cluster infrastructure (choose [aws, azure, vsphere, gcp, preprovisioned]) and the name of the cluster.
+Upgrade the core addons in a cluster using the 'dkp upgrade addons' command specifying the cluster infrastructure (choose [aws, azure, vsphere, gcp, eks, preprovisioned]) and the name of the cluster.  In the examples below, replace infrastructure provider in command at the location with corresponding provider name as shown in brackets above.
```
Comment (msdolbey) [inline pages/dkp/konvoy/2.3/dkp-upgrade/upgrade-konvoy/essential/index.md:103]:
```suggestion
Upgrade the core addons in a cluster using the `dkp upgrade addons` command specifying the cluster infrastructure (choose [`aws`, `azure`, `vsphere`, `gcp`, `eks`, `preprovisioned`]) and the name of the cluster. In the examples below, replace infrastructure provider in command at the location with corresponding provider name as shown in brackets above.
```

File: pages/dkp/konvoy/2.3/dkp-upgrade/upgrade-konvoy/essential/index.md [hunk 7]
```diff
@@ -156,7 +160,7 @@ When upgrading the Kubernetes version of a cluster, first upgrade the control pl
     - If an AMI was specified when initially creating a cluster for AWS, you must build a new one with <a href="/dkp/konvoy/2.3/image-builder/create-ami/">Konvoy Image Builder</a>.
     - If an Azure Machine Image was specified for Azure, you must build a new one with <a href="/dkp/konvoy/2.3/image-builder/create-azure-image/">Konvoy Image Builder</a>.
 
-1.  Upgrade the Kubernetes version of the control plane.
+1.  Upgrade the Kubernetes version of the control plane.  As shown in update addons section above, replace infrastructure provider name in the command. [aws, azure, vsphere, gcp, eks, preprovisioned].
```
Comment (msdolbey) [inline pages/dkp/konvoy/2.3/dkp-upgrade/upgrade-konvoy/essential/index.md:163]:
```suggestion
1.  Upgrade the Kubernetes version of the control plane. As shown in update addons section above, replace infrastructure provider name in the command [`aws`, `azure`, `vsphere`, `gcp`, `eks`, `preprovisioned`].
```

File: pages/dkp/konvoy/2.3/dkp-upgrade/upgrade-konvoy/essential/index.md [hunk 8]
```diff
@@ -170,14 +174,14 @@ When upgrading the Kubernetes version of a cluster, first upgrade the control pl
      ✓ Updating the control plane
     ```
 
-1.  Upgrade the Kubernetes version of each of your node pools. Replace `my-nodepool` with the name of the node pool.
+1.  Upgrade the Kubernetes version of each of your node pools. Replace `my-nodepool` with the name of the node pool.  Also, replace the name of the infrastructure provider in the command [aws, azure, vsphere, gcp, eks, preprovisioned].
```
Comment (msdolbey) [inline pages/dkp/konvoy/2.3/dkp-upgrade/upgrade-konvoy/essential/index.md:177]:
```suggestion
1.  Upgrade the Kubernetes version of each of your node pools. Replace `my-nodepool` with the name of the node pool. Also, replace the name of the infrastructure provider in the command [`aws`, `azure`, `vsphere`, `gcp`, `eks`, `preprovisioned`].
```

File: pages/dkp/konvoy/2.3/dkp-upgrade/upgrade-konvoy/enterprise/index.md [hunk 9]
```diff
@@ -202,3 +206,4 @@ For the overall process for upgrading to the latest version of DKP, refer back t
 [backup]: ../../../../../kommander/2.3/backup-and-restore#back-up-on-demand
 [envariables2]: ../../../choose-infrastructure/aws/quick-start-aws#configure-aws-prerequisites
 [envariables3]: ../../../choose-infrastructure/vsphere/new#create-a-new-vsphere-kubernetes-cluster
+[envariables4]: ../../../choose-infrastructure/eks/quick-start-eks#configure-eks-prerequisites
```
Comment (msdolbey) [inline pages/dkp/konvoy/2.3/dkp-upgrade/upgrade-konvoy/enterprise/index.md:209]:
```suggestion
[envariables4]: ../../../choose-infrastructure/eks/quick-start#configure-eks-prerequisites
```

File: pages/dkp/konvoy/2.3/dkp-upgrade/upgrade-konvoy/essential/index.md [hunk 10]
```diff
@@ -200,3 +204,4 @@ For the overall process for upgrading to the latest version of DKP, refer back t
 [envariables]: ../../../choose-infrastructure/azure/quick-start-azure#configure-azure-prerequisites
 [envariables2]: ../../../choose-infrastructure/aws/quick-start-aws#configure-aws-prerequisites
 [envariables3]: ../../../choose-infrastructure/vsphere/new#create-a-new-vsphere-kubernetes-cluster
+[envariables4]: ../../../choose-infrastructure/eks/quick-start-eks#configure-eks-prerequisites
```
[the remark this answers is not in the document]
Comment (msdolbey) [reply pages/dkp/konvoy/2.3/dkp-upgrade/upgrade-konvoy/essential/index.md:207]: I don't know why - but github UI freaked out and had to select all three of them - only the last line needs a change.

**Review by msdolbey — APPROVED** [review]
Some nits - but looks good to me.

Also, you'll probably want to pull again to pick up the change in the accept.txt file if you have more changes. But it'll probably be okay, because it's in a totally different file @mmcelroy91 

nice

**Review by gvasquezvargas — APPROVED** [review]

File: pages/dkp/konvoy/2.3/dkp-upgrade/upgrade-konvoy/essential/index.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (mmcelroy91) [reply pages/dkp/konvoy/2.3/dkp-upgrade/upgrade-konvoy/essential/index.md:207]: Thank you sir!

File: pages/dkp/konvoy/2.3/choose-infrastructure/gcp/quick-start-gcp/index.md (hunk 2 above)
Comment (mmcelroy91) [reply pages/dkp/konvoy/2.3/choose-infrastructure/gcp/quick-start-gcp/index.md:27]: Well honestly, this is what was in Dimitri's notes, so I better verify this before I merge.

Comment (mmcelroy91) [reply pages/dkp/konvoy/2.3/choose-infrastructure/gcp/quick-start-gcp/index.md:27]: So yeah no, that was backend stuff, not customer facing, so I removed clusterctl reference.

**Pull request merged**